src/parser/jobs/smn/modules/Ruin2.js

```diff
@@ -60,8 +59,8 @@ export default class Ruin2 extends Module {
 		// Calc the time in the GCD that the boss can't be targeted - R2ing before an invuln to prevent an R3 cancel is good
 		const invulnTime = this.invuln.getUntargetableUptime(
 			'all',
+			this._lastGcd.timestamp,
 			event.timestamp,
-			event.timestamp + this.gcd.getEstimate(),
```

Comment:
this was a weird one, enough that I was doubting myself.
previous logic was checking untarget for the current GCD _forward_, to permit R2 on the _previous_ GCD. This fixes it to check untarget time for the GCD that was actually an R2.

Reply:
This change looks correct, though I was also doubting it at first as well.